File: 900/B.ArrayCloningTechnique.cpp
#include <iostream>
#include <unordered_map>
#include <vector>
#include <algorithm>
using namespace std;

int main() {
    int t;
    cin >> t; // Read the number of test cases
    while(t--) {
        int n;
        cin >> n; // Read the number of elements
        vector<int> numbers(n);
        unordered_map<int, int> d;

        for(int i = 0; i < n; i++) {
            cin >> numbers[i];
        }

        int m = 0;
        for(int i : numbers) {
            d[i]++;
            m = max(m, d[i]);
        }

        int count = 0;
        int value = m;
        while (value < n) {
            count++;
            if (2 * value > n) {
                count += n - value;
                break;
            } else {
                count += value;
                value = 2 * value;
            }
        }

        cout << count << endl;
    }

    return 0;
}

File: 900/B.ArrayCloningTechnique.py
t=int(input())
for _ in range(t):
    n=int(input())
    numbers=list(map(int,input().split()))
    d={}
    m=0
    for i in numbers:
        if i in d:
            d[i]+=1
        else:
            d[i]=1
        m=max(m,d[i])
    count=0
    value=m
    while value<n:
        count+=1 
        if 2*value>n:
            count+=n-value
            break
        else:
            count+=value
            value=2*value
    print(count)

    